operator/src/main/java/org/bf2/operator/operands/KafkaCluster.java

```diff
@@ -207,11 +210,11 @@ public Kafka kafkaFrom(ManagedKafka managedKafka, Kafka current) {
                 .editOrNewSpec()
                     .editOrNewKafka()
                         .withVersion(this.kafkaManager.currentKafkaVersion(managedKafka))
-                        .withConfig(buildKafkaConfig(managedKafka, current))
+                        .withConfig(buildKafkaConfig(managedKafka, current, storagePerBroker))
                         .withReplicas(actualReplicas)
                         .withResources(config.kafka.buildResources())
                         .withJvmOptions(buildKafkaJvmOptions(managedKafka))
-                        .withStorage(buildKafkaStorage(managedKafka, current))
+                        .withStorage(buildKafkaStorage(managedKafka, current, storagePerBroker, formatOverheadWithMargin))
```

Comment:
To avoid the downward travelling flag should we calculate the `safetyMargin` and `formatOverheadWithMargin` in `buildKafkaStorage`

Reply:
Agreed